pkg/provider/provider.go

```diff
@@ -195,7 +195,9 @@ func buildTestEnvironment() { //nolint:funlen,gocyclo
 		} else {
 			env.AntiAffinityRequiredPods = append(env.AntiAffinityRequiredPods, &aNewPod)
 		}
-		env.Containers = append(env.Containers, getPodContainers(&pods[i])...)
+
+		// Build slices of containers and containers specifically not ignored globally.
+		env.Containers = append(env.Containers, getPodContainers(&pods[i], true)...)
```

Comment:
Seems like the container list in the Pod object is not filtered while the container list itself is filtered. Seems confusing to me. Shouldn't we have a dedicated filtered object?

Reply:
Yes I did this on purpose.  The Service Mesh test uses `containers` derived from the pod itself, but the `env.Containers` is used everywhere else.  So I added a boolean to `getPodContainers()` to be able to skip the filtering for containers stored in the `Pod` object, but apply the filter for the overall list.

Reply:
right, I am jut saying that it might be confusing to someone going through the container list and expect it to match the pod's container list since there is no indication that the container list is filtered (other than looking at the code).

Reply:
This sounds like a good compromise.  I'll add a comment on the struct.